### PR TITLE
Fixed some annotations

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

- accept int or array for where selects
- accept Phrase for notifications
- accept int for loadByAttribute

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @sreichel. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=19758) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=19758) before we can accept your contribution.<br/><hr/>**Sven Reichel** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

File: app/code/Magento/Catalog/Model/AbstractModel.php [hunk 1]
```diff
@@ -235,7 +235,7 @@ public function getResourceCollection()
      * Load entity by attribute
      *
      * @param \Magento\Eav\Model\Entity\Attribute\AttributeInterface|integer|string|array $attribute
-     * @param null|string|array $value
+     * @param null|string|int|array $value
```
Comment (rogyar) [inline app/code/Magento/Catalog/Model/AbstractModel.php]: Can we have `float` here as well? 
Please, take a look at `addAttributeToFilter` method of the same class. The `$condition` parameter should have the same types as hinted for `$value` in `loadByAttribute()`. The parameter also can have int and float types. So, the annotation should be adjusted there as well. Thanks

File: lib/internal/Magento/Framework/DB/Select.php [hunk 2]
```diff
@@ -101,7 +101,7 @@ public function __construct(
      * </code>
      *
      * @param string $cond The WHERE condition.
-     * @param string $value OPTIONAL A single value to quote into the condition.
+     * @param string|int|array|null $value OPTIONAL A single value to quote into the condition.
```
Comment (rogyar) [inline lib/internal/Magento/Framework/DB/Select.php]: Can be `float` as well.

**Review by rogyar — CHANGES_REQUESTED** [review]
Hi @sreichel. Thank you for your collaboration. Please, consider fixing the minor issues described in the review

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=19758) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=19758) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: sreichel<br/>:x: Sven Reichel<hr/>**Sven Reichel** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

**Comment by sreichel** [comment]
Please ignore CLA .... used wrong account.

**Comment by sidolov** [comment]
Hi @sreichel , please, force push your changes with correct account, we can't accept PR without signed CLA by all contributors. 
Thank you!

**Comment by sidolov** [comment]
Hi @sreichel , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**